contrib_bots/bots/weather/weather.py

```diff
@@ -0,0 +1,69 @@
+# See readme.md for instructions on running this code.
+import requests
+import json
+import os
+import sys
+from six.moves.configparser import SafeConfigParser
+
+
+DIRECTORY = os.path.dirname(os.path.realpath(__file__))
+if not os.path.exists(DIRECTORY + '/.weather_config'):
+    print('Weather bot config file not found, please set up it in /.weather_config'
+          '\n\nUsing format:\n\n[weather-config]\nkey=<OpenWeatherMap API key here>\n\n')
+    sys.exit(1)
+
+class WeatherHandler(object):
+    def usage(self):
+        return '''
+            This plugin will give info about weather in specified city
+            '''
+
+
+    def handle_message(self, message, client, state_handler):
+        help_content = '''
+            Info on Zulip can be found here:
+            https://github.com/zulip/zulip
+            '''.strip()
+
+        print(message)
+
+        if message['content'] == 'help':
+            response = help_content
+        else:
+            url = 'http://api.openweathermap.org/data/2.5/weather?q=' + message['content'] + '&APPID='
+            r = requests.get(url+get_weather_api_key_from_config())
+            print r.text
+            if "city not found" in r.text:
+                response = "Sorry, city not found"
+            else:
+                response = format_text(r.text, message['content'])
+
+
+        client.send_message(dict(
+            type='stream',
+            to=message['display_recipient'],
+            subject=message['subject'],
+            content=response,
+        ))
+
+def format_text(text, city):
+    j = json.loads(text)
+    return "Weather in " + j['name'] + ', ' + j['sys']['country'] + ':\n' \
+        + str(to_fahrenheit(j['main']['temp'])) + ' F / ' \
+        + str(to_celsius(j['main']['temp'])) + ' C\n' \
+        + j['weather'][0]['description'].title() + '\n'
```

Comment:
Do you think it would be much cleaner to do something like this?

```
response_pattern = 'Weather in {}, {}:\n{} F / {} C\n{}'
return response_pattern.format(city, country, fahrenheit, celsius, description)
```